src/System.Net.Security/tests/FunctionalTests/NegotiateStreamTestForUnix.cs

```diff
@@ -453,6 +475,295 @@ public void NegotiateStream_StreamToStream_KerberosAuthInvalidTarget_Failure()
+
+            }
+        }
+
+
+        [Theory, OuterLoop]
+        [MemberData("ValidNtlmCredentials")]
+        [PlatformSpecific(PlatformID.Linux)]
+        public void NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success(NetworkCredential credential)
+        {
+            if (!_isNtlmAvailable)
+            {
+                _output.WriteLine("skipping NegotiateStream_StreamToStream_NtlmAuthentication_ValidCredentials_Success");
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new NegotiateStream(clientStream))
+            using (var server = new UnixGssFakeNegotiateStream(serverStream))
+            {
+                Assert.False(client.IsAuthenticated, "client is not authenticated");
+                Assert.False(server.IsAuthenticated, "server is not authenticated");
+
+                Task[] auth = new Task[2];
+                auth[0] = client.AuthenticateAsClientAsync(credential, _testTarget, ProtectionLevel.None, TokenImpersonationLevel.Identification);
+                auth[1] = server.AuthenticateAsServerAsync();
+
+                bool finished = Task.WaitAll(auth, TestConfiguration.PassingTestTimeoutMilliseconds);
+                Assert.True(finished, "Handshake completed in the allotted time");
+
+                // Expected Client property values:
+                Assert.True(client.IsAuthenticated, "client is now authenticated");
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.False(client.IsEncrypted, "client is not encrypted");
+                Assert.False(client.IsMutuallyAuthenticated, "client is mutually authenticated");
+                Assert.False(client.IsServer, "client is not server");
+                Assert.True(client.IsSigned, "client is signed");
+                Assert.False(client.LeaveInnerStreamOpen, "inner stream remains open");
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("NTLM", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated, "server identity is authenticated");
+                IdentityValidator.AssertHasName(serverIdentity, _testTarget);
+            }
+        }
+
+     
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux)]
+        public void NegotiateStream_StreamToStream_NtlmAuthentication_EmptyTarget_KerberosUser_Fallback_Success()
+        {
+            if (!_isNtlmAvailable)
+            {
+                _output.WriteLine("skipping NegotiateStream_StreamToStream_NtlmAuthentication_EmptyTarget_KerberosUser_Fallback_Success");
```

Comment:
I don't understand why these aren't `[ConditionalFact(nameof(IsNtlmAvailable))]`, which removes the ifs.